Private/Invoke-Scans.ps1

```diff
@@ -0,0 +1,107 @@
+function Invoke-Scans {
+    [CmdletBinding()]
+    param (
+    # Could split Scans and PromptMe into separate parameter sets.
+    [Parameter()]
+        [ValidateSet('Auditing','ESC1','ESC2','ESC3','ESC4','ESC5','ESC6','ESC8','All','PromptMe')]
+        [array]$Scans = 'All'
+    )
+
+    # Is this needed?
+    if ($Scans -eq $IsNullOrEmpty) {
+        $Scans = 'All'
+    }
+
+    if ( $Scans -eq 'PromptMe' ) {
+        $GridViewTitle = 'Select the tests to run and press Enter or click OK to continue...'
+
+        # Check for Out-GridView or Out-ConsoleGridView
+        if ((Get-Command Out-ConsoleGridView -ErrorAction SilentlyContinue) -and ($PSVersionTable.PSVersion.Major -ge 7)) {
+            [array]$Scans = ($Dictionary | Select-Object Name,Type,Category | Out-ConsoleGridView -OutputMode Multiple -Title $GridViewTitle).Name | Sort-Object -Property Name
+        }
+        elseif (Get-Command -Name Out-GridView -ErrorAction SilentlyContinue) {
+            [array]$Scans = ($Dictionary | Select-Object Name,Type,Category | Out-GridView -PassThru -Title $GridViewTitle).Name | Sort-Object -Property Name
+        }
+        else {
+            # To Do: Check for admin and prompt to install features/modules or revert to 'All'.
+            Write-Information "Out-GridView and Out-ConsoleGridView were not found on your system. Defaulting to `'All`'."
+            $Scans = 'All'
+        }
+    }
+
+    switch ( $Scans ) {
+        Auditing {
+            Write-Host 'Identifying auditing issues...'
+            [array]$AuditingIssues = Find-AuditingIssue -ADCSObjects $ADCSObjects
+        }
+        ESC1 {
+            Write-Host 'Identifying AD CS templates with dangerous configurations...'
+            [array]$ESC1 = Find-ESC1 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
+        }
+        ESC2 {
+            Write-Host 'Identifying AD CS templates with dangerous configurations...'
+            [array]$ESC2 = Find-ESC2 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
+        }
+        ESC3 {
+            Write-Host 'Identifying AD CS templates with dangerous configurations...'
+            [array]$ESC3 = Find-ESC3Condition1 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
+            [array]$ESC3 += Find-ESC3Condition2 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
+        }
+        ESC4 {
+            Write-Host 'Identifying AD CS template and other objects with poor access control...'
+            [array]$ESC4 = Find-ESC4 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers -DangerousRights $DangerousRights -SafeOwners $SafeOwners
+        }
+        ESC5 {
+            Write-Host 'Identifying AD CS template and other objects with poor access control...'
+            [array]$ESC5 = Find-ESC5 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers -DangerousRights $DangerousRights -SafeOwners $SafeOwners
+        }
+        ESC6 {
+            Write-Host 'Identifying AD CS template and other objects with poor access control...'
```

Comment:
Add `ESC6` into this `Write-Host`.. OR maybe this title could be improved to talk about the dumb flag?

Reply:
Added ESC6 for now. Maybe we can work on getting better descriptions and taxonomy in the dictionary and then pull that into relevant parts of the script.